src/peft/peft_model.py

```diff
@@ -388,18 +397,33 @@ def load_adapter(self, model_id, adapter_name, is_trainable=False, **kwargs):
         # load weights if any
         path = os.path.join(model_id, kwargs["subfolder"]) if kwargs.get("subfolder", None) is not None else model_id
 
-        if os.path.exists(os.path.join(path, WEIGHTS_NAME)):
+        # load_safetensors is only used for remote weights
+        remote_weights_name = SAFETENSORS_WEIGHTS_NAME if load_safetensors else WEIGHTS_NAME
```

Comment:
Since the default is `False` this will always use the pickled file iiuc unless user specifically asks for it.